frontend/src/components/widgets/CameraImageInput/CameraImageInput.tsx

```diff
@@ -0,0 +1,499 @@
+    const fileId = this.state.files[this.state.files.length - 1].id
+    this.deleteFile(fileId)
+
+    this.setState({
+      imgSrc: null,
+    })
+  }
+
+  public onMediaError(): void {
+    this.setState({
+      webcamRequestState: "error",
+    })
+  }
+
+  public onUserMedia(): void {
+    this.setState({
+      webcamRequestState: "success",
+    })
+  }
+
+  get initialValue(): State {
+    const emptyState = {
+      files: [],
+      newestServerFileId: 0,
+      imgSrc: null,
+      webcamRequestState: "pending",
+    }
+    const { widgetMgr, element } = this.props
+
+    const widgetValue = widgetMgr.getFileUploaderStateValue(element)
+    if (widgetValue == null) {
+      return emptyState
+    }
+
+    const { maxFileId, uploadedFileInfo } = widgetValue
+    if (maxFileId == null || maxFileId === 0 || uploadedFileInfo == null) {
+      return emptyState
+    }
+
+    return {
+      imgSrc: null,
+      webcamRequestState: "pending",
+      files: uploadedFileInfo.map(f => {
+        const name = f.name as string
+        const size = f.size as number
+        const serverFileId = f.id as number
+
+        return new UploadFileInfo(name, size, this.nextLocalFileId(), {
+          type: "uploaded",
+          serverFileId,
+        })
+      }),
+      newestServerFileId: Number(maxFileId),
+    }
+  }
+
+  public componentWillUnmount(): void {
+    this.formClearHelper.disconnect()
+  }
+
+  /**
+   * Return the FileUploader's current status, which is derived from
+   * its state.
+   */
+  public get status(): FileUploaderStatus {
+    const isFileUpdating = (file: UploadFileInfo): boolean =>
+      file.status.type === "uploading"
+
+    // If any of our files is Uploading or Deleting, then we're currently
+    // updating.
+    if (this.state.files.some(isFileUpdating)) {
+      return "updating"
+    }
+
+    return "ready"
+  }
+
+  public componentDidUpdate = (prevProps: Props): void => {
+    const { element, widgetMgr } = this.props
+
+    // Widgets are disabled if the app is not connected anymore.
+    // If the app disconnects from the server, a new session is created and users
+    // will lose access to the files they uploaded in their previous session.
+    // If we are reconnecting, reset the file uploader so that the widget is
+    // in sync with the new session.
+    if (prevProps.disabled !== this.props.disabled && this.props.disabled) {
+      this.reset()
+      widgetMgr.setFileUploaderStateValue(
+        element,
+        new FileUploaderStateProto(),
+        { fromUi: false }
+      )
+      return
+    }
+
+    // Maybe send a widgetValue update to the widgetStateManager.
+
+    // If our status is not "ready", then we have uploads in progress.
+    // We won't submit a new widgetValue until all uploads have resolved.
+    if (this.status !== "ready") {
+      return
+    }
+
+    // If we have had no completed uploads, our widgetValue will be
+    // undefined, and we can early-out of the state update.
```

Comment:
i think this comment should maybe be 
"and we can stop the status update early" or something like that.